packages/@o3r/components/src/rules-engine/placeholder.rules-engine.effect.ts

```diff
@@ -88,17 +107,20 @@ export class PlaceholderTemplateResponseEffect {
       set[fact.name] = fact.factValue;
       return set;
     }, {});
-    const replacements$: Observable<{ ejsVar: RegExp; value: string }>[] = [];
+    const replacements$: Observable<{ ejsVar: RegExp; value: string } | null>[] = [];
     if (vars && template) {
       for (const varName in vars) {
         if (Object.prototype.hasOwnProperty.call(vars, varName)) {
           const ejsVar = new RegExp(`<%=\\s*${varName}\\s*%>`, 'g');
           switch (vars[varName].type) {
             case 'relativeUrl': {
               replacements$.push(
-                this.dynamicContentService.getMediaPathStream(vars[varName].value).pipe(
-                  take(1),
-                  map((value: string) => ({ejsVar, value}))
+                from(this.dynamicContentService).pipe(
```

Comment:
dynamicContentService seems to be a promise, shouldn't it be resolved only once ? (returning the synchronous injector.get) Maybe I misunderstood something about the dynamic import, but why do we need to pipe a switchmap here ?

Reply:
Your understanding is correct, the promise from `this.dynamicContentService` does resolve only once.
The tricky point here is that we are not in an async function, and we do not really want to store the result of the `dynamicContentService` promise in an private field (to avoid `undefined` case issue).

So actually the code
```typescript
replacements$.push(
  from(dynamicContentService).pipe(switch((service) => service.observable)
)
```
can be written:
```typescript
replacements$.push(
  from(
    dynamicContentService.then((service) => service.observable) // Promise<Observable<>>
  ).switch((observable) => observable)
)
```
but it sound more complicated (at least to me) :S.